src/TerminalFighter/Terminal/Terminal.cpp

```diff
@@ -36,9 +36,41 @@ void Terminal::draw(I_GraphicsHandler& graphics) {
         set_texture(graphics.load_image(terminal_texture_path_));
     }
 
+    SDL_Color white = {255, 255, 255};
     SDL_Surface* UIText = TTF_RenderText_Blended(default_font_, player_text_.c_str(), TF_Colors::WHITE);
-    graphics.draw(terminal_texture_, (int)x_pos(), (int)y_pos(), GraphicPriority::UI, true);
-    graphics.draw(UIText, (int)x_pos() + 30, (int)y_pos() + 30, GraphicPriority::UI, true);
+    SDL_Rect ui_src_rect;
+    ui_src_rect.x = 0;
+    ui_src_rect.y = 0;
+
+    if (UIText == NULL) {
+        ui_src_rect.w = 0;
+        ui_src_rect.h = 0;
+    } else {
+        ui_src_rect.w = UIText->w;
+        ui_src_rect.h = UIText->h;
+    }
+
+    SDL_Rect terminal_src_rect;
+    SDL_QueryTexture(terminal_texture_, NULL, NULL, &terminal_src_rect.w, &terminal_src_rect.h);
+    terminal_src_rect.x = 0;
+    terminal_src_rect.y = 0;
```

Comment:
My biggest concern is having to create this src_rect for every class. Everytime we have a new object we have this same ugly 4 lines of code.

Reply:
sorry, i did not replace this with the new fix.
The code for example on basicEnemy.cpp is:

```
    graphics.draw(texture_,
                  graphics.create_default_rect(texture_),
                  (int)x_pos(),
                  (int)y_pos(),
                  GraphicPriority::MIDDLE,
                  true,
                  0,
                  NULL);
 }
 }
```

Reply:
Kk I saw this in a few places. After this is fixed we can merge.

Reply:
Can we write unit tests for the animator class?